ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/common/block/AbstractBlockProcessor.java

```diff
@@ -606,13 +607,37 @@ public void processDeposits(MutableBeaconState state, SszList<? extends Deposit>
       throws BlockProcessingException {
     safelyProcess(
         () -> {
+          final boolean depositSignaturesAreAllGood = batchVerifyDepositSignatures(deposits);
           for (Deposit deposit : deposits) {
-            processDeposit(state, deposit);
+            processDeposit(state, deposit, depositSignaturesAreAllGood);
           }
         });
   }
 
-  public void processDeposit(MutableBeaconState state, Deposit deposit) {
+  private boolean batchVerifyDepositSignatures(SszList<? extends Deposit> deposits) {
+    final Bytes32 domain = miscHelpers.computeDomain(Domain.DEPOSIT);
+    final List<List<BLSPublicKey>> publicKeys = new ArrayList<>();
+    final List<Bytes> messages = new ArrayList<>();
+    final List<BLSSignature> signatures = new ArrayList<>();
+    for (Deposit deposit : deposits) {
+      final BLSPublicKey pubkey = deposit.getData().getPubkey();
+      final Bytes signingRoot =
+          miscHelpers.computeSigningRoot(
+              new DepositMessage(
+                  pubkey,
+                  deposit.getData().getWithdrawalCredentials(),
+                  deposit.getData().getAmount()),
+              domain);
```

Comment:
nit: Probably worth extracting a method to calculate the signingRoot for a `Deposit` as we're doing it here and in the existing `depositSignatureIsValid` and they always have to be the same.

Reply:
👍